src/proof_system/quotient_poly.rs

```diff
@@ -41,14 +42,15 @@ pub(crate) fn compute(
     w4_eval_4n.push(w4_eval_4n[3]);
 
     let t_1 = compute_circuit_satisfiability_equation(
-        domain,
+        &domain_4n,
         preprocessed_circuit,
         (&wl_eval_4n, &wr_eval_4n, &wo_eval_4n, &w4_eval_4n),
         public_inputs_poly,
     );
 
     let t_2 = compute_permutation_checks(
         domain,
+        &domain_4n,
```

Comment:
It's cheap to create Evaluation domains, so instead of adding extra parameters, lets revert it back to just passing in the domain of size `n` and then the local function can decide how it wants to deal with it.

This way does have the drawback that those methods will return a result, but I think the advantage of encapsulating all of the logic in one function is a better benefit.

Reply:
I still don't see which is the problem of sending them.

At the end, the difference is 6 against 5 params in one function and on the second one is exactly the same number, but you send the  correct domain directly.

I really don't see where's the problem here tbh

Reply:
Either way is acceptable. I opted for having a single domain parameter, so that all logic for that specific computation is encapsulated in the function.

Each function receives the same domain of size `n` and can choose to expand it depending on the logic that is needed inside of the function . Since computing a domain is cheap, and it is allocated on the stack, there is no performance impact to just re-computing it locally and throwing it away.

Currently, there are only two domain sizes, however when we also had a domain of size `2n`, this was the most debuggable version, as you cannot mistakenly pass a domain of size `2n` as `4n`.

Reply:
kk. Will revert that.